core/internal/mocks/eth_client.go

```diff
@@ -2,27 +2,32 @@
 
 package mocks
 
-import assets "chainlink/core/store/assets"
-import big "math/big"
-import common "github.com/ethereum/go-ethereum/common"
-import ethereum "github.com/ethereum/go-ethereum"
-import mock "github.com/stretchr/testify/mock"
-import models "chainlink/core/store/models"
-
-// EthClient is an autogenerated mock type for the EthClient type
-type EthClient struct {
+import (
+	assets "chainlink/core/assets"
+	big "math/big"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	common "github.com/ethereum/go-ethereum/common"
+
+	mock "github.com/stretchr/testify/mock"
+
+	eth "chainlink/core/eth"
+)
+
+// Client is an autogenerated mock type for the Client type
+type Client struct {
 	mock.Mock
 }
 
 // GetBlockByNumber provides a mock function with given fields: hex
-func (_m *EthClient) GetBlockByNumber(hex string) (models.BlockHeader, error) {
+func (_m *Client) GetBlockByNumber(hex string) (eth.BlockHeader, error) {
```

Comment:
There may have been a mistake here with the renaming, as we're no longer in the package 'eth'

Reply:
This is a special case because of `mocks`. All `mocks` of types found in other packages are placed here.

Reply:
Oh I understand.  I mean should it still be called 'EthClient' here? because the package name isn't eth.

Reply:
Conceptually I see where you're coming from. But this is autogenerated code originating from `mockery`, and its convention is for the mock to have the same name as the interface. We use `go generate` to produce it, but even if I use the `mockery` CLI, I don't think there's an option to change the name of the output type:

```
Usage of mockery:
  -all
    	generates mocks for all found interfaces in all sub-directories
  -case string
    	name the mocked file using casing convention [camel, snake, underscore] (default "camel")
  -cpuprofile string
    	write cpu profile to file
  -dir string
    	directory to search for interfaces (default ".")
  -inpkg
    	generate a mock that goes inside the original package
  -keeptree
    	keep the tree structure of the original interface files into a different repository. Must be used with XX
  -name string
    	name or matching regular expression of interface to generate mock for
  -note string
    	comment to insert into prologue of each generated file
  -outpkg string
    	name of generated package (default "mocks")
  -output string
    	directory to write mocks to (default "./mocks")
  -print
    	print the generated mock to stdout
  -quiet
    	suppress output to stdout
  -recursive
    	recurse search into sub-directories
  -tags string
    	space-separated list of additional build tags to use
  -testonly
    	generate a mock in a _test.go file
  -version
    	prints the installed version of mockery
```

`-name` is for the origin interface, not the output mock.